### PR TITLE
Fix Temperature Update and Gradient Norm Clipping

### PR DESCRIPTION
**Pull request by SharpKoi**

### CODE REVIEW

**Comment by ToooooooT** [comment]
gradient clipping on alpha mean and alpha std not alpha!
